File: my-code/75/main.go
package main

import "fmt"

// @Time     : 2021/08/06 09:07
// @Author   : Ranshi
// @File     : main.go

func sortColors(nums []int) {
	start, end := 0, len(nums)-1
	for idx := 0; idx <= end; idx++ {
		if nums[idx] == 0 {
			nums[start], nums[idx] = nums[idx], nums[start]
			start++
		} else if nums[idx] == 2 {
			nums[end], nums[idx] = nums[idx], nums[end]
			end--
			if nums[idx] != 1 {
				idx--
			}
		}
	}
}

func main() {
	nums := []int{2, 0, 2, 1, 1, 0}
	sortColors(nums)
	fmt.Printf("%v\n", nums)
}

File: my-code/75/main.py
# -*- coding:utf-8 -*-
# @Time     : 2021/05/28 17: 53
# @Author   : Ranshi
# @File     : 75. 颜色分类.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums[:] = nums.count(0) * [0] + nums.count(1) * [1] + nums.count(2) * [2]


if __name__ == "__main__":
    s = Solution()
    s.sortColors(nums=[2, 0, 2, 1, 1, 0])
